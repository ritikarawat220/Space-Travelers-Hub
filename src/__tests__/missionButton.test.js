import React from 'react';
import renderer from 'react-test-renderer';
import MissionButton from '../components/missionButton';

describe('MissionButton', () => {
  const onClickMock = jest.fn();

  it('renders correctly when not reserved', () => {
    const component = renderer.create(
      <MissionButton isReserved={false} onClick={onClickMock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when reserved', () => {
    const component = renderer.create(
      <MissionButton isReserved onClick={onClickMock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the onClick function when clicked', () => {
    const component = renderer.create(
      <MissionButton isReserved={false} onClick={onClickMock} />,
    );
    const button = component.root.findByType('button');
    button.props.onClick();
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
