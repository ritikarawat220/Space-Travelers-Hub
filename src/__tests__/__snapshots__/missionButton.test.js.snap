// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionButton renders correctly when not reserved 1`] = `
<button
  className="button joinMission"
  onClick={[MockFunction]}
  type="button"
>
  Join Mission
</button>
`;

exports[`MissionButton renders correctly when reserved 1`] = `
<button
  className="button leaveMission"
  onClick={[MockFunction]}
  type="button"
>
  Leave Mission
</button>
`;
